### PR TITLE
[ragile] adapter for kernel 5.x

### PR DESCRIPTION
**Pull request by pettershao-ragilenetworks**
Signed-off-by: “pettershao” <pettershao@ragilenetworks.com>

#### Why I did it
Ragile adapter ra-b6510-32c ra-b6510-48v8c ra-b6910-64c ra-b6920-4s to kernel 5.x
#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111

### CODE REVIEW

**Comment by pettershao-ragilenetworks** [comment]
@sujinmkang It's refolked for https://github.com/Azure/sonic-buildimage/pull/9942 because my local repo rebase faile, please merge this one, thanks!

**Comment by pettershao-ragilenetworks** [comment]
why build vs canceled?

**Comment by pettershao-ragilenetworks** [comment]
@sujinmkang help merge again, vs build cancelled abnormal last time, thanks!

**Comment by pettershao-ragilenetworks** [comment]
@sujinmkang @lguohan help merge this, thanks!

File: platform/broadcom/sonic-platform-modules-ragile/ra-b6510-32c/modules/driver/pddf_custom_led_module.c [hunk 1]
```diff
@@ -303,16 +323,24 @@ ssize_t set_status_led(struct device_attribute *da)
...
+#endif
         pddf_dbg(LED, KERN_INFO "Set color:%s; 0x%x:0x%x sys_val:0x%x new_val:0x%x read:0x%x\n",
 		LED_STATUS_STR[cur_state],
                 ops_ptr->swpld_addr, ops_ptr->swpld_addr_offset,
                 sys_val, new_val,
-		ret = board_i2c_cpld_read(ops_ptr->swpld_addr, ops_ptr->swpld_addr_offset));
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 10, 0)
+        ret = board_i2c_cpld_read(ops_ptr->swpld_addr, ops_ptr->swpld_addr_offset));
+#else
+		ret = board_i2c_cpld_read_new(ops_ptr->swpld_addr, ops_ptr->device_name, ops_ptr->swpld_addr_offset));
```
Comment (sujinmkang) [inline platform/broadcom/sonic-platform-modules-ragile/ra-b6510-32c/modules/driver/pddf_custom_led_module.c:342]: change the tab to spaces.

**Review by sujinmkang — APPROVED** [review]

**Comment by sujinmkang** [comment]
@pettershao-ragilenetworks can you please also rebase this PR too?

File: platform/pddf/platform-modules-pddf.mk [hunk 2]
```diff
@@ -12,4 +12,5 @@ SONIC_DPKG_DEBS += $(PDDF_PLATFORM_MODULE)
 PDDF_PLATFORM_MODULE_SYM = sonic-platform-pddf-sym_$(PDDF_PLATFORM_MODULE_VERSION)_amd64.deb
 $(eval $(call add_derived_package,$(PDDF_PLATFORM_MODULE),$(PDDF_PLATFORM_MODULE_SYM)))
 
+
```
[the remark this answers is not in the document]
Comment (pettershao-ragilenetworks) [reply platform/pddf/platform-modules-pddf.mk:15]: this change is for rebuild pddf, or may build failed:
```
Module.symvers dependency dropped.
WARNING: Symbol version dump "/sonic/platform/broadcom/sonic-platform-modules-ragile/ra-b6510-32c/../../../pddf/i2c/Module.symvers.PDDF" is missing.
         Modules may not have dependencies or modversions.
  MODPOST /sonic/platform/broadcom/sonic-platform-modules-ragile/ra-b6510-32c/modules/driver/Module.symvers
```

[the remark this answers is not in the document]
Comment (pettershao-ragilenetworks) [reply platform/pddf/platform-modules-pddf.mk]: it is used for rebuild pddf for fix below issue:
```

make[5]: Circular /sonic/platform/broadcom/sonic-platform-modules-ragile/ra-b6510-32c/modules/driver/Module.symvers <- /sonic/platform/broadcom/sonic-platform-modules-ragile/ra-b6510-32c/modules/driver/Module.symvers dependency dropped.
WARNING: Symbol version dump "/sonic/platform/broadcom/sonic-platform-modules-ragile/ra-b6510-32c/../../../pddf/i2c/Module.symvers.PDDF" is missing.
         Modules may not have dependencies or modversions.
  MODPOST /sonic/platform/broadcom/sonic-platform-modules-ragile/ra-b6510-32c/modules/driver/Module.symvers
ERROR: modpost: "get_device_i2c_kobj" [/sonic/platform/broadcom/sonic-platform-modules-ragile/ra-b6510-32c/modules/driver/pddf_custom_led_module.ko] undefined!

```

**Comment by pettershao-ragilenetworks** [comment]
> @pettershao-ragilenetworks can you please also rebase this PR too?

hi, @sujinmkang all done ,help forward it, thanks!

**Comment by pettershao-ragilenetworks** [comment]
hi, @sujinmkang all done ,help forward it, thanks!

**Review by sujinmkang — APPROVED** [review]
looks good to me.

**Comment by pettershao-ragilenetworks** [comment]
> looks good to me.

done cla!!

**Comment by pettershao-ragilenetworks** [comment]
@lguohan can help merge this, thanks!

**Comment by pettershao-ragilenetworks** [comment]
@sujinmkang  @lguohan  help merge this, thanks!

**Comment by pettershao-ragilenetworks** [comment]
@yxieca Please help merge this, thanks!

**Comment by pettershao-ragilenetworks** [comment]
@yxieca @sujinmkang help check if any thing need modify, I think building fail is caused by other reason not changed file. Help forward this to merge state, thanks, waiting your reply.

**Comment by pettershao-ragilenetworks** [comment]
[MergePR_checklog.txt](https://github.com/sonic-net/sonic-buildimage/files/10357918/MergePR_checklog.txt)
@yxieca @sujinmkang @lguohan have fix all updating issue, help merge this, thanks!

**Review by yxieca — APPROVED** [review]

**Pull request merged**